Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript

```diff
@@ -2455,30 +2455,35 @@ command cancelPendingMessages
    end if
 end cancelPendingMessages
 
-on selectionChanged pArrowKey
+-- PM-2016-09-22: [[ Bug 15887 ]] Distinguish between "selectionChanged" sent by the engine and \
+-- "selectionChangedByArrowKey pArrowKey"
+on selectionChanged
    if the selectedText of field "Script" of me is not empty then
       put the selectedText of field "Script" of me into sLastNonEmptySelection
    end if
    
-   get the selectedChunk
-   
    # OK-2008-10-22 : Bug 7343 - Must save the last selection point as its lost
    # when the user begins typing into the search field.
    if the long id of the focusedObject is the long id of field "Script" of me then
       put word 2 of it & comma & word 4 of it into sLastSelectedChunk
    end if
+end selectionChanged
+
+on selectionChangedByArrowKey pArrowKey
+   send "selectionChanged" to me in 0 milliseconds
```

Comment:
@livecodepanos whenever we find ourselves sending engine messages I think it's a good time to re-evaluate what we are doing. My suggestion is to move the selection changed handling to a separate handler and send that when required.